dm/syncer/dbconn/upstream_db.go

```diff
@@ -71,9 +71,9 @@ func (conn *UpStreamConn) GetServerUnixTS(ctx context.Context) (int64, error) {
 	return utils.GetServerUnixTS(ctx, conn.BaseDB.DB)
 }
 
-// GetCharsetAndDefaultCollation returns charset and default collation map.
-func (conn *UpStreamConn) GetCharsetAndDefaultCollation(ctx context.Context) (map[string]string, error) {
-	return utils.GetCharsetAndDefaultCollation(ctx, conn.BaseDB.DB)
+// GetCharsetAndDefaultCollation returns charset and collation info.
+func (conn *UpStreamConn) GetCharsetAndCollationInfo(ctx context.Context) (map[string]string, map[int]string, error) {
+	return utils.GetCharsetAndCollationInfo(ctx, conn.BaseDB.DB)
```

Comment:
ok, but maybe i will have to send query in `dm/syncer/dbconn/upstream_db.go` , because `github.com/pingcap/ticdc/dm/pkg/conn` and `github.com/pingcap/ticdc/dm/pkg/utils` will have `import cycle not allowed` error.

Reply:
i had this problem once too... 😢